src/AppBundle/Service/EventProcessor.php

```diff
@@ -811,4 +817,20 @@ private function logEnd(string $key, ?string $message = null): void
         $duration = round($this->stopwatch->getEvent($key)->getDuration());
         $this->log(" <comment>$message ($duration ms)</comment>");
     }
+
+    /**
+     * Checks every database connection to see if it still works, and reconnects them if needed.
+     */
+    private function reconnectDatabases(): void
+    {
+        /** @var Registry $registry */
+        $registry = $this->container->get('doctrine');
+        /** @var Connection $connection */
+        foreach ($registry->getConnections() as $connection) {
+            if (!$connection->ping()) {
+                $connection->close();
+                $connection->connect();
+            }
```

Comment:
Why not move these checks to each connection method in `Repository`? https://github.com/wikimedia/eventmetrics/blob/master/src/AppBundle/Repository/Repository.php#L197-L259

There are other scenarios where this could happen, such as the reports. Anything that has a sequence of long-running queries.

Reply:
I will say this is the exact approach I took before, and it didn't work for whatever reason. Hopefully it does this time!